pkg/apis/pgv2.percona.com/v2/perconapgcluster_types.go

```diff
@@ -47,6 +47,7 @@ type PerconaPGCluster struct {
 	Status PerconaPGClusterStatus `json:"status,omitempty"`
 }
 
+// +kubebuilder:validation:XValidation:rule="!has(self.users) || self.postgresVersion >= 15 || self.users.all(u, !has(u.grantPublicSchemaAccess) || !u.grantPublicSchemaAccess)",message="PostgresVersion must be >= 15 if grantPublicSchemaAccess exists and is true"
```

Comment:
I believe we can introduce an envtest that asserts that we will get the expected failure message if someone tries to configure grantPublicSchemaAccess with a version other than 15.